relay-server/src/actors/store.rs

```diff
@@ -839,13 +849,116 @@ impl StoreService {
                 return Ok(());
             }
         };
+
+        let Some(span) = annotated_span.value() else {
```

Comment:
This should make the `.clone()`s below unnecessary.

```suggestion
        let Some(span) = annotated_span.into_value() else {
```